api/pkg/s3/uploadpart.go

```diff
@@ -115,6 +127,16 @@ func (s *APIService) UploadPart(request *restful.Request, response *restful.Resp
 		return
 	}
 
+	// Check if sha256sum match.
+	if needCheckSha256 {
```

Comment:
Can this part of the code be moved up at lin#99?

Reply:
No, data is transfered through a datastream, from line#96 to line#100, it's just init the sha256Writer, that means make the datastream go through the sha256Writer, once all data transfered, we can use sha256Writer to calculate the sha256sum. And from line#103 and line#121 is for the real data transfer, that means we have to do the sum calculation after that.

Reply:
Ok. Thanks